### PR TITLE
Implement both -isEqual: and -isEqualToString: to directly call the internal _isEqual, to avoid a redundant ObjC message send + type check

### PR DESCRIPTION
**Pull request by Catfish-Man**

### CODE REVIEW

**Comment by Catfish-Man** [comment]
@swift-ci please smoke test

**Comment by Catfish-Man** [comment]
@swift-ci please smoke benchmark

**Comment by Catfish-Man** [comment]
@lorentey reviewed in person :)

File: stdlib/public/core/StringStorage.swift [hunk 1]
```diff
@@ -303,6 +303,12 @@ final internal class __StringStorage
 
   @objc(isEqualToString:)
   @_effects(readonly)
+  final internal func isEqualToString(to other: AnyObject?) -> Int8 {
```
Comment (jrose-apple) [inline stdlib/public/core/StringStorage.swift]: I feel like you could do a bit better on the Swift name here :-P

**Pull request merged**